### PR TITLE
G-API: Make GFrame a new (distinct) G-type, not an alias to GMat

### PR DESCRIPTION
**Pull request by dmatveev**
- The underlying host type is still cv::Mat, a new cv::MediaFrame
  type is to be added as a separate PR.

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2020.4.0:16.04
build_image:Custom Win=openvino-2020.4.0
build_image:Custom Mac=openvino-2020.4.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/src/api/gframe.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return *m_priv;
+}
+
+const cv::GOrigin& cv::GFrame::priv() const {
+    return *m_priv;
+}
+
+namespace cv {
+std::ostream& operator<<(std::ostream& os, const cv::GFrameDesc &desc) {
+    (void) desc;
```
Comment (TolyaTalamanov) [inline modules/gapi/src/api/gframe.cpp:38]: Just skip `desc` in argument list

File: modules/gapi/test/gapi_frame_tests.cpp [hunk 2]
```diff
@@ -17,15 +17,13 @@ G_API_OP(GBlurFrame, <GMat(GFrame)>, "test.blur_frame") {
     }
 };
 
-GAPI_OCV_KERNEL(OCVBlurFrame, GBlurFrame)
```
Comment (TolyaTalamanov) [inline modules/gapi/test/gapi_frame_tests.cpp:20]: Here and below. 
Why did you change the brackets ?

**Review by TolyaTalamanov — APPROVED** [review]

File: modules/gapi/include/opencv2/gapi/gcommon.hpp [hunk 3]
```diff
@@ -80,6 +80,7 @@ enum class GShape: int
     GSCALAR,
     GARRAY,
     GOPAQUE,
+    GFRAME,
```
Comment (TolyaTalamanov) [inline modules/gapi/include/opencv2/gapi/gcommon.hpp:83]: Make sense to handle it in corresponding `switch`

**Comment by TolyaTalamanov** [comment]
Thanks! LGTM 👍

File: modules/gapi/test/gapi_frame_tests.cpp (hunk 2 above)
Comment (dmatveev) [reply modules/gapi/test/gapi_frame_tests.cpp:20]: Consistency reasons - look at this file

File: modules/gapi/include/opencv2/gapi/gcommon.hpp (hunk 3 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gcommon.hpp:83]: it is already handled in `default:` which throws.

File: modules/gapi/src/api/gframe.cpp (hunk 1 above)
Comment (dmatveev) [reply modules/gapi/src/api/gframe.cpp:38]: Good

File: modules/gapi/include/opencv2/gapi/gcommon.hpp (hunk 3 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/gcommon.hpp:83]: haha, no. Added now

**Comment by dmatveev** [comment]
@alalek @mshabunin can we proceed with merge?

**Comment by smirnov-alexey** [comment]
Looks good. Does it have to do something with the recently introduced `RMat`?

**Comment by dmatveev** [comment]
> Does it have to do something with the recently introduced RMat?

A common spirit, I believe. Nothing else :)
